scripts/o3de/o3de/manifest.py

```diff
@@ -195,23 +197,25 @@ def load_o3de_manifest(manifest_path: pathlib.Path = None) -> dict:
             return json_data
 
 
-def save_o3de_manifest(json_data: dict, manifest_path: pathlib.Path = None) -> bool:
+def save_o3de_manifest(json_data: dict, manifest_path: pathlib.Path = None) -> None:
     """
-        Save the json dictionary to the supplied manifest file or ~/.o3de/o3de_manifest.json if manifest_path is None
+    Save the json dictionary to the supplied manifest file or ~/.o3de/o3de_manifest.json if None
 
-        :param json_data: dictionary to save in json format at the file path
-        :param manifest_path: optional path to manifest file to save
-        """
+    :param json_data: dictionary to save in json format at the file path
+    :param manifest_path: optional path to manifest file to save
+    """
     if not manifest_path:
         manifest_path = get_o3de_manifest()
+    backup_name = utils.backup_file(manifest_path)
     with manifest_path.open('w') as s:
         try:
             s.write(json.dumps(json_data, indent=4) + '\n')
-            return True
-        except OSError as e:
+        except Exception as e:
```

Comment:
Globally catching Exception is bad practice.
a try block should only catch the exceptions it can handle.
Furthermore the logic here should be to write to a temporary file and then move it to the `manifest_path` file.
That way a backup file isn't needed